apps/meteor/app/meteor-accounts-saml/server/lib/SAML.ts

```diff
@@ -220,8 +221,8 @@ export class SAML {
 			},
 		);
 
-		if (username && username !== user.username) {
-			await saveUserIdentity({ _id: user._id, username });
+		if (username && fullName && (username !== user.username || fullName !== user.name)) {
```

Comment:
If there's no username or no fullName, the other will also not be saved.

Reply:
oops, thanks for noticing this :grimacing: please check #30177